File: USACO/Bronze/sleepinginclass.java
import java.io.*;
import java.util.*;

public class sleepinginclass {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		PrintWriter pw = new PrintWriter(System.out);
		StringTokenizer st = new StringTokenizer(br.readLine());
		int t = Integer.parseInt(st.nextToken());
		for (int tt = 0; tt < t; tt++) {
			st = new StringTokenizer(br.readLine());
			int n = Integer.parseInt(st.nextToken()), sum = 0, ans = n - 1;
			ArrayList<Integer> arr = new ArrayList<Integer>();
			st = new StringTokenizer(br.readLine());
			for (int i = 0; i < n; i++) {
				int x = Integer.parseInt(st.nextToken());
				sum+= x;
				arr.add(x);
			}
			if (sum == 0) {
				pw.println(0);
				continue;
			}
			ArrayList<Integer> divs = new ArrayList<Integer>();
			for (int i = 1; i * i <= sum; i++) {
				if (sum % i == 0) {
					divs.add(i);
					divs.add(sum / i);
				}
			}
			for (int div : divs) {
				int part = 0, cnt = 0;
				for (int i = 0; i < n; i++) {
					part+= arr.get(i);
					cnt++;
					if (part > div) {
						cnt = n - 1;
						break;
					}
					else if (part == div) {
						part = 0;
						cnt--;
					}
				}
				ans = Math.min(ans, cnt);
			}
			pw.println(ans);
		}
		pw.close();
	}
}

File: USACO/Bronze/sleepinginclass.cpp
#include <bits/stdc++.h>
#define mp make_pair
#define sz(x) (int) (x).size() 
#define pb push_back
 
using namespace std;
 
typedef long long ll;
 
const int MAXN = 5e8 + 5;
const int MOD = 998244353;
const int INF = 0x3f3f3f3f;
const ll INFLL = 0x3f3f3f3f3f3f3f3f;
const double EPS = 1e-6;

void solve() {
	int n, sum = 0; cin >> n;
	int ans = n - 1;
	vector<int> arr(n);
	for (int i = 0; i < n; i++) {
		cin >> arr[i];
		sum+= arr[i];
	}
	if (sum == 0) {
		cout << 0 << '\n';
		return;
	}
	vector<int> divs;
	for (int i = 1; i * i <= sum; i++) {
		if (sum % i == 0) {
			divs.pb(i);
			divs.pb(sum / i);
		}
	}
	for (int &div : divs) {
		int part = 0, cnt = 0;
		for (int i = 0; i < n; i++) {
			part+= arr[i];
			cnt++;
			if (part > div) {
				cnt = n - 1;
				break;
			}
			else if (part == div) {
				part = 0;
				cnt--;
			}
		}
		ans = min(ans, cnt);
	}
	cout << ans << '\n';
}

int main() {
	ios::sync_with_stdio(0); cin.tie(0);
	int t; cin >> t;
	while (t--) {
		solve();
	}
	return 0;
}